lib/fluent/plugin_helper/http_server/compat/server.rb

```diff
@@ -24,16 +25,26 @@ module Compat
         class Server
           # @param logger [Logger]
           # @param default_app [Object] ignored option. only for compat
-          def initialize(addr:, port:, logger:, default_app: nil)
+          # @param tls_context [OpenSSL::SSL::SSLContext]
+          def initialize(addr:, port:, logger:, default_app: nil, tls_context: nil)
             @addr = addr
             @port = port
             @logger = logger
-            @server = WEBrick::HTTPServer.new(
+
+            config = {
               BindAddress: @addr,
               Port: @port,
               Logger: WEBrick::Log.new(STDERR, WEBrick::Log::FATAL),
               AccessLog: [],
-            )
+            }
+            if tls_context
+              require 'webrick/https'
+              @logger.warn('Webrick ignores given TLS version')
```

Comment:
Does this bug of webrick or spec?

Reply:
Spec. webrick does not provide an interface of setting the TLS version.
See https://github.com/ruby/webrick/blob/c5635fa5e2d4b7d5ca2e08e4c1ae6ec13c7237ec/lib/webrick/ssl.rb#L180-L205